marconi-sidechain/src/Marconi/Sidechain/Api/Routes.hs

```diff
@@ -102,21 +102,44 @@ type RpcEpochNonceMethod =
 -- Query and Result types
 --------------------------
 
-newtype GetCurrentSyncedBlockResult
-  = GetCurrentSyncedBlockResult (WithOrigin BlockInfo)
+newtype SidechainTip = SidechainTip {getTip :: C.ChainTip}
+  deriving (Eq, Ord, Generic, Show)
+
+instance ToJSON SidechainTip where
+  toJSON (SidechainTip C.ChainTipAtGenesis) = Aeson.Null
+  toJSON (SidechainTip (C.ChainTip sn bhh bn)) =
+    Aeson.object
+      [ "blockNo" .= bn
+      , "blockHeaderHash" .= bhh
+      , "slotNo" .= sn
+      ]
+
+instance FromJSON SidechainTip where
+  parseJSON Aeson.Null = pure $ SidechainTip C.ChainTipAtGenesis
+  parseJSON obj =
+    let parseTip v = do
+          slotNo <- v .: "slotNo"
+          bhh <- v .: "blockHeaderHash"
+          bn <- v .: "blockNo"
+          pure $ SidechainTip $ C.ChainTip slotNo bhh bn
+     in Aeson.withObject "ChainTip" parseTip obj
+
+data GetCurrentSyncedBlockResult
+  = GetCurrentSyncedBlockResult (WithOrigin BlockInfo) SidechainTip
   deriving (Eq, Ord, Generic, Show)
 
 instance ToJSON GetCurrentSyncedBlockResult where
-  toJSON (GetCurrentSyncedBlockResult blockInfoM) =
+  toJSON (GetCurrentSyncedBlockResult blockInfoM tip) =
     let chainPointObj = case blockInfoM of
           (At (BlockInfo sn bhh bn bt en)) ->
             [ "blockNo" .= bn
             , "blockTimestamp" .= bt
             , "blockHeaderHash" .= bhh
             , "slotNo" .= sn
             , "epochNo" .= en
+            , "nodeTip" .= toJSON tip
             ]
-          Origin -> []
+          Origin -> ["nodeTip" .= toJSON tip]
```

Comment:
For consistency, how about *not* setting the "nodeTip" field like the rest of the fields in this scenario? It will just be null anyway.